experimental/unitsconversion/src/measureunit.rs

```diff
@@ -13,14 +13,18 @@ use crate::{
     ConversionError,
 };
 
-// TODO(#4369): split this struct to two structs: MeasureUnitParser for parsing the identifier and MeasureUnit to represent the unit.
-// TODO NOTE: the MeasureUnitParser takes the trie and the ConverterFactory takes the full payload and an instance of MeasureUnitParser.
-pub struct MeasureUnit {
-    /// Contains the processed units.
-    pub contained_units: SmallVec<[MeasureUnitItem; 8]>,
+/// A parser for the CLDR unit identifier (e.g. `meter-per-square-second`)
+pub struct MeasureUnitParser<'data> {
+    /// Contains the zero-trie payload.
+    zerotrie_payload: &'data ZeroTrie<ZeroVec<'data, u8>>,
```

Comment:
nit: don't repeat type information in names

Reply:
done

Reply:
also in the field name

Reply:
done